### PR TITLE
Feature: Support eth_getTransactionCount rpc call

### PR DESCRIPTION
**Pull request by mikheevm**
This PR:
- add support for `eth_getTransactionCount` rpc call in sdk and provider

### CODE REVIEW

File: .changeset/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changesets
```
Comment (rmeissner) [inline .changeset/README.md:1]: This feels unrelated to adding support for `eth_getTransactionCount`

Comment (mikheevm) [reply .changeset/README.md:1]: This is auto-generated initialization files by versioning manager we use. I forgot to add it in the corresponding pr

File: packages/safe-apps-web3modal/package.json [hunk 2]
```diff
@@ -29,7 +29,6 @@
     "web3modal": "^1.9.3"
   },
   "peerDependencies": {
-    "@gnosis.pm/safe-apps-sdk": "4.0.0",
```
Comment (germartinez) [inline packages/safe-apps-web3modal/package.json:32]: Shouldn't this be a dependency instead of a devDependency?
It is imported here: https://github.com/gnosis/safe-apps-sdk/blob/master/packages/safe-apps-web3modal/src/modal.ts#L1

File: .changeset/config.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@1.6.0/schema.json",
```
Comment (germartinez) [inline .changeset/config.json:2]: I guess this attribute was autogenerated? It does not appear on the docs https://github.com/atlassian/changesets/blob/main/docs/config-file-options.md but has the same structure.

**Review by germartinez — APPROVED** [review]

File: packages/safe-apps-web3modal/package.json (hunk 2 above)
Comment (mikheevm) [reply packages/safe-apps-web3modal/package.json:32]: there's currently a mess going on with dependencies 🤦 , it's already a dependency of a dependency package

Comment (mikheevm) [reply packages/safe-apps-web3modal/package.json:32]: I'll just remove this change and keep it like it was because it worked 😄

File: .changeset/config.json (hunk 3 above)
Comment (mikheevm) [reply .changeset/config.json:2]: Yes
